src/script/miniscript.h

```diff
@@ -523,6 +523,35 @@ struct Node {
         }
     }
 
+    NodeRef<Key> Clone() const
+    {
+        // Use TreeEval() to avoid a stack-overflow due to recursion
+        auto upfn = [](const Node& node, Span<NodeRef<Key>> children) {
+            NodeRef<Key> ret;
+            // As all members of Node are const, except for subs, we need to construct the cloned node with all of these members.
+            // However, there is no constructor that takes all three of data, keys, and subs.
+            // But, they are mutually exclusive, so we can use the appropriate constructor depending on what is available.
+            if (!node.keys.empty()) {
+                Assert(node.data.empty() && node.subs.empty());
+                ret = MakeNodeRef<Key>(internal::NoDupCheck{}, node.m_script_ctx, node.fragment, node.keys, node.k);
+            } else if (!node.data.empty()) {
+                Assert(node.keys.empty() && node.subs.empty());
+                ret = MakeNodeRef<Key>(internal::NoDupCheck{}, node.m_script_ctx, node.fragment, node.data, node.k);
+            } else if (!node.subs.empty()) {
```

Comment:
just an observation: in all branches other than this one `children.empty() == true` since `node.subs.empty() == true`, and more generally `children.size() == node.subs.size()`

Verified that this is the case with the following diff:

```diff
diff --git a/src/script/miniscript.h b/src/script/miniscript.h
@@ -527,6 +527,7 @@ struct Node {
     {
         // Use TreeEval() to avoid a stack-overflow due to recursion
         auto upfn = [](const Node& node, Span<NodeRef<Key>> children) {
+            Assert(node.subs.size() == children.size());
             NodeRef<Key> ret;
```